src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -6711,6 +6762,63 @@ components:
           items:
             type: string
 
+    SnapshotBuilderAccessRequest:
+      type: object
+      properties:
+        name:
+          type: string
+        researchPurposeStatement:
+          type: string
+        datasetRequest:
+          $ref: '#/components/schemas/SnapshotBuilderRequest'
+      description: Request for access to a snapshot.
+
+    SnapshotBuilderRequest:
+      type: object
+      properties:
+        cohorts:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderCohort'
+        conceptSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderDatasetConceptSets'
+        valueSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderFeatureValueGroup'
+      description: Specification of requested a snapshot, returned in SnapshotBuilderAccessRequest.
+
+    SnapshotBuilderCohort:
+      type: object
+      properties:
+        name:
+          type: string
+        criteriaGroups:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderCriteriaGroup'
+      description: Selection of participants.
+
+    SnapshotBuilderCriteriaGroup:
+      type: object
+      properties:
+        name:
+          type: string
+        criteria:
+          type: array
+          items:
+            oneOf:
+              - $ref: '#/components/schemas/SnapshotBuilderDomainOption'
```

Comment:
I think you may need a new type for this. The `Option` versions of these types are like "prototype" or "source" versions. For a selected domain option, the only value needed is the option's concept code. The Domain root (which I think the UI calls `category`) may also be helpful to include for non-OMOP cases. For OMOP data the code is unique across all categories so it's sufficient by itself.

Note that a future version of the cohort builder will support other constraints on the codes, like occurrence count and occurrence times.